frontend/src/components/file-explorer/TreeNode.test.tsx

```diff
@@ -1,148 +1,121 @@
 import React from "react";
-import { act, render } from "@testing-library/react";
+import { act } from "@testing-library/react";
 import userEvent from "@testing-library/user-event";
+import { renderWithProviders } from "test-utils";
 import TreeNode from "./TreeNode";
-import { WorkspaceFile } from "#/services/fileService";
-
-const onFileClick = vi.fn();
-
-const NODE: WorkspaceFile = {
-  name: "folder",
-  children: [
-    { name: "file.ts" },
-    { name: "folder2", children: [{ name: "file2.ts" }] },
-  ],
-};
+import { selectFile, listFiles } from "#/services/fileService";
+
+vi.mock("../../services/fileService", async () => ({
+  listFiles: vi.fn(async (path: string = "/") => {
+    if (path === "/") {
+      return Promise.resolve(["folder1/", "file1.ts"]);
+    }
+    if (path === "/folder1/") {
+      return Promise.resolve(["file2.ts"]);
+    }
+    return Promise.resolve([]);
+  }),
+  selectFile: vi.fn(async () => Promise.resolve({ code: "Hello world!" })),
+  uploadFile: vi.fn(),
+}));
 
 describe("TreeNode", () => {
   afterEach(() => {
     vi.resetAllMocks();
   });
```

Comment:
```suggestion
  beforeEach(() => {
    vi.clearAllMocks();
  });
```

`.resetAllMocks` resets all mock implementations you have defined above to their default state.
`.clearAllMocks` will only clear the mock state, but not reset the implementation you have defined above.

In your case, you want to clear the mocks since you want to use your implementation in other tests.

Reply:
Thanks!